packages/core/src/provider/base.ts

```diff
@@ -56,6 +57,14 @@ export const noOpMiddleware: AccountMiddlewareFn = async (
 ) => struct;
 
 export interface SmartAccountProviderOpts {
+  /**
+   * Optional entry point contract address for override if needed.
+   * If not provided, the default entry point contract for the chain will be used.
+   * Refer to https://docs.alchemy.com/reference/eth-supportedentrypoints for all the supported entrypoints
+   * when using Alchemy as your RPC provider.
+   */
+  entryPointAddress?: Address;
```

Comment:
hmm, why can't we just make the provider's already existing `entryPointAddress` parameter be optional with a default value of `getDefaultEntryPointContract(chain)`?

Reply:
very loose prototype:

```
class BaseProvider {
  constructor({
    chain,
    entryPointAddress = getDefaultEntryPointAddress(chain),
  }: {
    chain: Chain;
    entryPointAddress?: Address;
  }) {
    console.log(entryPointAddress);
  }
}
```

^this type checks for me and I can access the entryPoint value directly as a non-optional type:

![image](https://github.com/alchemyplatform/aa-sdk/assets/43521356/332eca7b-da5c-47ff-8740-742c1dbb6c88)

Reply:
+1 good catch, this is a bit cleaner